pkg/ingester/ingester.go

```diff
@@ -1002,67 +1052,83 @@ func (i *Ingester) startPushRequest(ctx context.Context, reqSize int64) (_ conte
 		rejectEqualInflightBytes = true // if inflightBytes == limit, reject new request
 	}
 
-	finishRequestInDefer := true
+	instanceLimitsErr := i.checkInstanceLimits(inflight, inflightBytes, rejectEqualInflightBytes)
+	if instanceLimitsErr == nil {
+		return ctx, true, nil
+	}
 
-	defer func() {
-		if finishRequestInDefer {
-			i.finishPushRequest(reqSize)
-		}
-	}()
+	// In this case a per-instance limit has been hit, and the corresponding error has to be passed
```

Comment:
Where exactly did we set error into `pushRequestState`?

Reply:
Thank you for finding this bug. I have fixed the implementation.